buildSrc/src/main/kotlin/com/virtuslab/gitmachete/buildsrc/UpdateIntellijVersions.kt

```diff
@@ -44,67 +44,68 @@ open class UpdateIntellijVersions : DefaultTask() {
   private fun findReleaseNewerThan(version: String): String? {
     return findFirstMatchingVersionNewerThan(
       intellijReleasesContents,
-      Regex("(?<=ideaIC-)(\\d+\\.)+\\d+(?=.pom)"),
+      Regex("""(?<=ideaIC-)(\d+\.)+\d+(?=.pom)"""),
       version
     )
   }
 
   private fun findLatestMinorOfVersion(versionNumber: String): String {
-    val major = getMajorPart(versionNumber)
+    val major = versionToMajorVersion(versionNumber)
 
     return findFirstMatchingVersionNewerThan(
       intellijReleasesContents,
-      Regex("(?<=ideaIC-)$major(\\.\\d+)?(?=.pom)"),
+      Regex("""(?<=ideaIC-)$major(\.\d+)?(?=.pom)"""),
       major
     ) ?: versionNumber
   }
 
   private fun findEapWithBuildNumberHigherThan(buildNumber: String): String? {
     return findFirstMatchingVersionNewerThan(
       intellijSnapshotsContents,
-      Regex("(?<=ideaIC-)\\d+\\.\\d+\\.\\d+(?=-EAP-SNAPSHOT.pom)"),
+      Regex("""(?<=ideaIC-)\d+\.\d+\.\d+(?=-EAP-SNAPSHOT\.pom)"""),
       buildNumber
     )
   }
 
   @TaskAction
   fun execute() {
     val intellijVersions: IntellijVersions by project.rootProject.extra
-    val properties = IntellijVersionHelper.getProperties()
-    val originalProperties = IntellijVersionHelper.getProperties()
-    val latestStable = intellijVersions.latestStable
-    val latestMinorsList = intellijVersions.latestMinorsOfOldSupportedMajors
-      .map { findLatestMinorOfVersion(it) }
+    val originalVersions = intellijVersions
+    var updatedVersions = originalVersions
```

Comment:
All updates are performed on `IntellijVersions`, and only at the end saved to `Properties` (the updates used to happen on `Properties` from the start, which was confusing)

Reply:
Other than that, no significant changes in logic